test/test_convolutional_layer.h

```diff
@@ -154,15 +154,15 @@ TEST(convolutional, fprop) {
   {
     l.forward_propagation(buf.in_buf(), buf.out_buf());
 
-    EXPECT_NEAR(0.4875026, out[0], 1E-5);
-    EXPECT_NEAR(0.8388910, out[1], 1E-5);
-    EXPECT_NEAR(0.8099984, out[2], 1E-5);
-    EXPECT_NEAR(0.7407749, out[3], 1E-5);
-    EXPECT_NEAR(0.5000000, out[4], 1E-5);
-    EXPECT_NEAR(0.1192029, out[5], 1E-5);
-    EXPECT_NEAR(0.5986877, out[6], 1E-5);
-    EXPECT_NEAR(0.7595109, out[7], 1E-5);
-    EXPECT_NEAR(0.6899745, out[8], 1E-5);
+    EXPECT_NEAR(float_t(-0.05), out[0], 1E-5);
+    EXPECT_NEAR(float_t(1.65), out[1], 1E-5);
+    EXPECT_NEAR(float_t(1.45), out[2], 1E-5);
+    EXPECT_NEAR(float_t(1.05), out[3], 1E-5);
+    EXPECT_NEAR(float_t(0.00), out[4], 1E-5);
+    EXPECT_NEAR(float_t(-2.0), out[5], 1E-5);
+    EXPECT_NEAR(float_t(0.40), out[6], 1E-5);
+    EXPECT_NEAR(float_t(1.15), out[7], 1E-5);
+    EXPECT_NEAR(float_t(0.80), out[8], 1E-5);
```

Comment:
@Randl yes. There was sigmoid earlier so the values were non recurring decimals, but since it is elementwise product of simple rational numbers, these values are rational.